src/Orchard.Cms.Web/Modules/Orchard.OpenId/Services/OpenIdConfiguration.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.IdentityModel.Tokens.Jwt;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using OpenIddict;
+using Orchard.OpenId.Services;
+using Orchard.OpenId.Settings;
+
+namespace Orchard.OpenId
+{
+    public class OpenIdConfiguration : IConfigureOptions<OpenIddictOptions>
+    {
+        private readonly ILogger<OpenIdConfiguration> _logger;
+        private readonly IOpenIdService _openIdService;
+
+        public OpenIdConfiguration(ILogger<OpenIdConfiguration> logger, IOpenIdService openIdService)
+        {
+            _logger = logger;
+            _openIdService = openIdService;
+        }
+
+        public void Configure(OpenIddictOptions options)
+        {   
+            var settings = _openIdService.GetOpenIdSettingsAsync().Result;
+            if (!_openIdService.IsValidOpenIdSettings(settings))
+                return;
+
+            if (settings.AccessTokenFormat == OpenIdSettings.TokenFormat.JWT)
+                options.AccessTokenHandler = new JwtSecurityTokenHandler();
+
+            if (settings.TestingModeEnabled)
+            {
+                options.SigningCredentials.AddEphemeralKey();
+                options.AllowInsecureHttp = true;
+            }
+            else if (settings.CertificateStoreLocation.HasValue && settings.CertificateStoreName.HasValue && !string.IsNullOrEmpty(settings.CertificateThumbPrint))
+            {
+                try
+                {
+
```

Comment:
It's not my code, I simply renamed the class to a simpler name. But I'll fix that (and the other remarks) :smile: